### PR TITLE
Detect log alert tx one 7449 v1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/7449

Describe changes:
- document more guess-applayer-tx
- improves guess-applayer-tx for unidirectional protocol such as DNS

### Provide values to any of the below to override the defaults.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2198

#12269 rebased with new SV test and new ticket in commit message

### CODE REVIEW

File: src/detect.c [hunk 1]
```diff
@@ -725,6 +725,28 @@ static inline void DetectRunPrefilterPkt(
 #endif
 }
 
+static bool isOnlyTxInDirection(Flow *f, uint64_t txid, uint8_t dir)
```
Comment (victorjulien) [inline src/detect.c:728]: style: `IsOnlyTxIsDirection`

Comment (victorjulien) [reply src/detect.c:728]: this needs a function doc as well

**Review by victorjulien — CHANGES_REQUESTED** [review]
Think the logic makes sense. A few comments inline.

File: src/detect.c (hunk 1 above)
Comment (catenacyber) [reply src/detect.c:728]: ok

**Comment by catenacyber** [comment]
Next PR is https://github.com/OISF/suricata/pull/12308

**Pull request closed without merging**